jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -242,14 +243,17 @@ def get_results(self):
 
         if not self.spectral_subset_valid:
             valid, spec_range, subset_range = self._check_dataset_spectral_subset_valid(return_ranges=True)  # noqa
-            raise ValueError(f"spectral subset '{self.spectral_subset.selected}' {subset_range} is outside data range of '{self.dataset.selected}' {spec_range}")  # noqa
+            raise ValueError(f"spectral subset '{self.spectral_subset.selected}' {subset_range}"
+                             f" is outside data range of '{self.dataset.selected}' {spec_range}")
 
         self._calculate_statistics()
         return self.results
 
     def _on_plotted_lines_changed(self, msg):
         self.line_marks = msg.marks
         self.line_items = msg.names_rest
+        menu_labels = [f"{msg.marks[i].name} {msg.marks[i].rest_value} {msg.marks[i].xunit}" for i in range(len(msg.marks))]  # noqa
+        self.line_menu_items = [{"title": menu_labels[i], "value": msg.names_rest[i]} for i in range(len(msg.names_rest))]  # noqa
```

Comment:
Might be a little long, but would be nice to avoid the double loop and index iterations.

```suggestion
        self.line_menu_items = [{"title": f"{mark.name} {mark.rest_value} {mark.xunit}", "value": name_rest} for mark, name_rest in zip(msg.marks, msg.names_rest)]  # noqa
```

Reply:
Fair point on the double loop, I erred on the side of not having one long line but performance is more important.